MetadataExtractor/IO/BufferReader.cs

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) Drew Noakes and contributors. All Rights Reserved. Licensed under the Apache License, Version 2.0. See LICENSE in the project root for license information.
+
+using System.Buffers.Binary;
+
+namespace MetadataExtractor.IO;
+
+internal ref struct BufferReader(ReadOnlySpan<byte> bytes, bool isBigEndian)
+{
+    private readonly ReadOnlySpan<byte> _bytes = bytes;
+    private int _position = 0;
+    private bool _isBigEndian = isBigEndian;
+
+    public int Position => _position;
+
+    public byte GetByte()
+    {
+        Debug.Assert(_position < _bytes.Length, "attempted to read past end of data");
```

Comment:
This assertion managed to sneak through.

Reply:
good catch.